src/col/vct/col/typerules/Types.scala

```diff
@@ -51,7 +51,8 @@ object Types {
     case (TType(left), TType(right)) =>
       TType(leastCommonSuperType(left, right))
 
-    case (TClass(left), TClass(right)) =>
+    case (TClass(left, Seq()), TClass(right, Seq())) =>
+      // TODO (RR): Not sure how generics factor into this part... See also: next match statement
```

Comment:
`transSupportArrows` should be defined on `TClass` and return a `Seq[(TClass[G], TClass[G])]`. We don't support {contra|co}variance yet (other than explicitly for axiomatic collections), so otherwise the type has to be equal.